### PR TITLE
android: Migrate remaining settings to ini

### PR DESCRIPTION
**Pull request by t895**
You can now change theme and input overlay settings directly in the config.ini file.

### Theme settings
image

### Overlay settings (it keeps going)
image

Additionally exposes a toggle for the touchscreen in Overlay options and updates to Kotlin 1.9.20 (needed for getting `entries` on an enum class)

Note - Button position for the overlay is represented as a number between 0 and 1. If you go beyond that range, the button will be drawn beyond the bounds of your screen.

### CODE REVIEW

File: src/yuzu/configuration/qt_config.cpp [hunk 1]
```diff
@@ -438,14 +441,14 @@ void QtConfig::SaveShortcutValues() {
...
-                     std::make_optional(default_hotkey.context));
-        WriteSetting(std::string("Repeat"), shortcut.repeat,
-                     std::make_optional(default_hotkey.repeat));
+        WriteStringSetting(std::string("KeySeq"), shortcut.keyseq,
+                           std::make_optional(default_hotkey.keyseq));
+        WriteStringSetting(std::string("Controller_KeySeq"), shortcut.controller_keyseq,
+                           std::make_optional(default_hotkey.controller_keyseq));
+        WriteIntegerSetting(std::string("Context"), shortcut.context,
+                            std::make_optional(default_hotkey.context));
+        WriteIntegerSetting(std::string("Repeat"), shortcut.repeat,
```
Comment (liamwhite) [inline src/yuzu/configuration/qt_config.cpp:450]:
```suggestion
        WriteBooleanSetting(std::string("Repeat"), shortcut.repeat,
```

**Review by liamwhite — CHANGES_REQUESTED** [review]
```
/__w/yuzu/yuzu/src/yuzu/configuration/qt_config.cpp:450: undefined reference to `std::enable_if<is_integral_v<bool>, void>::type Config::WriteIntegerSetting<bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool const&, std::optional<bool> const&, std::optional<bool> const&)'
```

**Review by liamwhite — APPROVED** [review]

**Comment by liamwhite** [comment]
Needs a rebase

**Comment by abouvier** [comment]
> ```
> /__w/yuzu/yuzu/src/yuzu/configuration/qt_config.cpp:450: undefined reference to `std::enable_if<is_integral_v<bool>, void>::type Config::WriteIntegerSetting<bool>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool const&, std::optional<bool> const&, std::optional<bool> const&)'
> ```

This error is still occuring.

```
[1897/1897] Linking CXX executable bin/yuzu
samu: job failed: : && /usr/lib/ccache/bin/c++ -march=znver2 -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -DNDEBUG -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now,-z,defs,--no-allow-shlib-undefined src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/mocs_compilation.cpp.o src/yuzu/CMakeFiles/yuzu.dir/about_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/applets/qt_amiibo_settings.cpp.o src/yuzu/CMakeFiles/yuzu.dir/applets/qt_controller.cpp.o src/yuzu/CMakeFiles/yuzu.dir/applets/qt_error.cpp.o src/yuzu/CMakeFiles/yuzu.dir/applets/qt_profile_select.cpp.o src/yuzu/CMakeFiles/yuzu.dir/applets/qt_software_keyboard.cpp.o src/yuzu/CMakeFiles/yuzu.dir/applets/qt_web_browser.cpp.o src/yuzu/CMakeFiles/yuzu.dir/bootmanager.cpp.o src/yuzu/CMakeFiles/yuzu.dir/compatibility_list.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configuration_shared.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_audio.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_camera.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_cpu.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_cpu_debug.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_debug.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_debug_controller.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_debug_tab.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_filesystem.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_general.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_graphics.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_graphics_advanced.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_hotkeys.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_input.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_input_advanced.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_input_per_game.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_input_player.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_input_player_widget.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_input_profile_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_mouse_panning.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_motion_touch.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_per_game.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_per_game_addons.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_profile_manager.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_ringcon.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_network.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_system.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_tas.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_touch_from_button.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_touchscreen_advanced.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_ui.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_vibration.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/configure_web.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/input_profiles.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/shared_translation.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/shared_widget.cpp.o src/yuzu/CMakeFiles/yuzu.dir/configuration/qt_config.cpp.o src/yuzu/CMakeFiles/yuzu.dir/debugger/console.cpp.o src/yuzu/CMakeFiles/yuzu.dir/debugger/controller.cpp.o src/yuzu/CMakeFiles/yuzu.dir/debugger/profiler.cpp.o src/yuzu/CMakeFiles/yuzu.dir/debugger/wait_tree.cpp.o src/yuzu/CMakeFiles/yuzu.dir/game_list.cpp.o src/yuzu/CMakeFiles/yuzu.dir/game_list_worker.cpp.o src/yuzu/CMakeFiles/yuzu.dir/hotkeys.cpp.o src/yuzu/CMakeFiles/yuzu.dir/install_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/loading_screen.cpp.o src/yuzu/CMakeFiles/yuzu.dir/main.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/chat_room.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/client_room.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/direct_connect.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/host_room.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/lobby.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/message.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/moderation_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/multiplayer/state.cpp.o src/yuzu/CMakeFiles/yuzu.dir/play_time_manager.cpp.o src/yuzu/CMakeFiles/yuzu.dir/qt_common.cpp.o src/yuzu/CMakeFiles/yuzu.dir/startup_checks.cpp.o src/yuzu/CMakeFiles/yuzu.dir/uisettings.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/clickable_label.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/controller_navigation.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/limitable_input_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/overlay_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/sequence_dialog/sequence_dialog.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/url_request_interceptor.cpp.o src/yuzu/CMakeFiles/yuzu.dir/util/util.cpp.o src/yuzu/CMakeFiles/yuzu.dir/vk_device_info.cpp.o src/yuzu/CMakeFiles/yuzu.dir/compatdb.cpp.o src/yuzu/CMakeFiles/yuzu.dir/qrc_languages.cpp.o src/yuzu/CMakeFiles/yuzu.dir/discord_impl.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/EWIEGA46WW/qrc_yuzu.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/IEX2ANJCOF/qrc_compatibility_list.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/JQYOR47KWS/qrc_controller.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/FPLSJTADT4/qrc_overlay.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/JJST62K4FD/qrc_style.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/QWG25RC7ZP/qrc_style.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/4OYJP4GYF3/qrc_style.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/L6763PLDTS/qrc_default.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/FCKMUTAFBG/qrc_style.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/3JV3SOUGRX/qrc_style.cpp.o src/yuzu/CMakeFiles/yuzu.dir/yuzu_autogen/37D3MJFPTG/qrc_style.cpp.o -o bin/yuzu  src/common/libcommon.a  src/core/libcore.a  src/input_common/libinput_common.a  src/frontend_common/libfrontend_common.a  src/network/libnetwork.a  src/video_core/libvideo_core.a  externals/glad/libglad.a  -lrt  /usr/lib/libQt5DBus.so.5.15.11  /usr/lib/libdiscord-rpc.so  /usr/lib/libQt5Multimedia.so.5.15.11  /usr/lib/libQt5WebEngineWidgets.so.5.15.16  /usr/lib/libusb-1.0.so  src/core/libcore.a  src/audio_core/libaudio_core.a  src/core/libcore.a  src/video_core/libvideo_core.a  src/audio_core/libaudio_core.a  externals/mbedtls/library/libmbedtls.a  externals/mbedtls/library/libmbedx509.a  externals/mbedtls/library/libmbedcrypto.a  src/network/libnetwork.a  src/web_service/libweb_service.a  src/network/libnetwork.a  src/web_service/libweb_service.a  /usr/lib/libenet.so  /usr/lib/libhttplib.so.0.14.3  /usr/lib/libbrotlicommon.so  /usr/lib/libbrotlienc.so  /usr/lib/libbrotlidec.so  /usr/lib/libz.so  /usr/lib/libssl.so  /usr/lib/libcrypto.so  externals/glad/libglad.a  -ldl  src/shader_recompiler/libshader_recompiler.a  externals/libbc_decoder.a  /usr/lib/libavcodec.so  /usr/lib/libavdevice.so  /usr/lib/libavfilter.so  /usr/lib/libavformat.so  /usr/lib/libavutil.so  /usr/lib/libswresample.so  /usr/lib/libswscale.so  -lva  externals/sirit/src/libsirit.a  src/common/libcommon.a  /usr/lib/libdl.a  /usr/lib/libboost_context.so.1.83.0  externals/libstb.a  /usr/lib/liblz4.so.1.9.4  /usr/lib/libzstd.so.1.5.5  externals/libdemangle.a  /usr/lib/libSDL2-2.0.so.0.2800.5  /usr/lib/libdynarmic.so.6.5.0  /usr/lib/libfmt.so.10.1.0  /usr/lib/libopus.so  /usr/lib/libcubeb.so.0.0.0  /usr/lib/libQt5WebEngineCore.so.5.15.16  /usr/lib/libQt5WebChannel.so.5.15.11  /usr/lib/libQt5Positioning.so.5.15.11  /usr/lib/libQt5Quick.so.5.15.11  /usr/lib/libQt5QmlModels.so.5.15.11  /usr/lib/libQt5Qml.so.5.15.11  /usr/lib/libQt5Network.so.5.15.11  /usr/lib/libQt5PrintSupport.so.5.15.11  /usr/lib/libQt5Widgets.so.5.15.11  /usr/lib/libQt5Gui.so.5.15.11  /usr/lib/libQt5Core.so.5.15.11 && :
/build/yuzu-git/src/yuzu-mainline/src/./video_core/texture_cache/formatter.h:14: warning: type ‘struct formatter’ violates the C++ One Definition Rule [-Wodr]
   14 | struct fmt::formatter<VideoCore::Surface::PixelFormat> : fmt::formatter<fmt::string_view> {
      | 
/build/yuzu-git/src/yuzu-mainline/src/./common/logging/formatter.h:14: note: a type with different bases is defined in another translation unit
   14 | struct fmt::formatter<T, std::enable_if_t<std::is_enum_v<T>, char>>
      | 
In function ‘copy’,
    inlined from ‘copy’ at /usr/include/c++/13.2.1/bits/char_traits.h:841:7,
    inlined from ‘_S_copy’ at /usr/include/c++/13.2.1/bits/basic_string.h:420:0,
    inlined from ‘_S_copy’ at /usr/include/c++/13.2.1/bits/basic_string.h:415:7,
    inlined from ‘_S_copy_chars’ at /usr/include/c++/13.2.1/bits/basic_string.h:474:0,
    inlined from ‘_M_construct’ at /usr/include/c++/13.2.1/bits/basic_string.tcc:243:0,
    inlined from ‘__ct ’ at /usr/include/c++/13.2.1/bits/basic_string.h:639:14,
    inlined from ‘__ct ’ at /usr/include/c++/13.2.1/bits/basic_string.h:631:7,
    inlined from ‘ShowNormalKeyboard’ at /build/yuzu-git/src/yuzu-mainline/src/core/frontend/applets/software_keyboard.cpp:68:21:
/usr/include/c++/13.2.1/bits/char_traits.h:850: warning: ‘__builtin_memcpy’ reading between 15 and 9223372036854775804 bytes from a region of size 10 [-Wstringop-overread]
  850 |                 (__builtin_memcpy(__s1, __s2, __n * sizeof(char_type))));
      | 
/usr/bin/ld: src/yuzu/CMakeFiles/yuzu.dir/configuration/qt_config.cpp.o: in function `QtConfig::SaveShortcutValues()':
qt_config.cpp:(.text+0x17ba): undefined reference to `std::enable_if<is_integral_v<int>, void>::type Config::WriteIntegerSetting<int>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int const&, std::optional<int> const&, std::optional<bool> const&)'
/usr/bin/ld: src/yuzu/CMakeFiles/yuzu.dir/configuration/qt_config.cpp.o: in function `QtConfig::SaveUIGamelistValues()':
qt_config.cpp:(.text+0xa7b4): undefined reference to `std::enable_if<is_integral_v<unsigned long>, void>::type Config::WriteIntegerSetting<unsigned long>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long const&, std::optional<unsigned long> const&, std::optional<bool> const&)'
collect2: error: ld returned 1 exit status
samu: subcommand failed
```

**Comment by liamwhite** [comment]
I am a bit confused about how WriteIntegerSetting<int> and WriteIntegerSetting<unsigned long> could be undefined in general when linking against frontend_common... but they do require explicit instantiations, because the template definition is not in the header.

**Pull request merged**